src/reifyhealth/specmonstah/core.clj

```diff
@@ -161,6 +161,73 @@
            attrs])
         query))
 
+(defn empty-ref?
+  [ref]
+  (and (vector? ref)
+       (empty? (second ref))
+       (empty? (last ref))))
+
+(defn- query-term-bindings
+  [query-term]
+  (->> (second query-term)
+       (medley/filter-keys vector?)
+       (medley/map-keys (fn [k] (if (= (count k) 2)
+                                 [(first k) ::template (second k)]
+                                 k)))))
+
+(defn- remove-query-term-bindings
+  [query-term]
+  (update query-term 1 (partial medley/filter-keys (complement vector?))))
+
+(defn- bind-term-relations
+  "Ensures that relations are consistent across a query term's tree"
+  ([relations query-term]
+   (let [bindings (query-term-bindings query-term)]
+     (if (empty? bindings)
+       query-term
+       (bind-term-relations
+         relations
+         (remove-query-term-bindings query-term)
+         bindings))))
+  ([relations [ent-type refs attrs query-ref-name generated?] bindings]
```

Comment:
It's from https://github.com/reifyhealth/specmonstah/pull/8/files#diff-1540ea4a039eb92498e5b62a633583e7R242 though I honestly can't remember why it's necessary